### PR TITLE
fix: Update CONTRIBUTING-docs.md with instructions for adding metadata to markdown files

### LINKED ISSUE #1097: [📝 Docs]:  Update docs contributing guide: "Add meta-data to newly created markdown file".
### What Docs changes are you proposing?

Could you update the [Docs contribution guide](https://github.com/json-schema-org/website/blob/main/CONTRIBUTING-docs.md) to include details of adding meta-data to every newly created markdown file? 
Every markdown page needs the meta-data information to run; this is an example;

```
---
title: 2019-09 Migration Notes
section: docs
---
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by techmannih**
**Issue Number:**
-  Closes #1097 

This PR updates the CONTRIBUTING-docs.md file to include instructions on adding metadata (YAML front matter) to new markdown files.
Added guidelines for including metadata fields like title, section, date, author, and tags.

### CODE REVIEW

File: CONTRIBUTING-docs.md [hunk 1]
```diff
@@ -64,6 +64,24 @@ After you have been assigned to an issue, fork the [json-schema-org/website](htt
 
 To propose modifications to our documentation that do not have an issue in the [documentation board](https://github.com/orgs/json-schema-org/projects/16), you can [create a GitHub issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-a-repository) and use the *Documentation* template to describe the changes. After that, you can decide whether you want to work on the changes yourself or let someone else claim the issue. If you decide to work on the issue, assign it to yourself and commit the changes to a new branch. 
 
+## Add Metadata to Newly Created Markdown Files
+When contributing to the documentation, it's important to include metadata in every newly created markdown file. This metadata helps organize content and ensure consistency across all pages in the documentation.When you create a new markdown file, add metadata to the file. The metadata should include the title of the page, the author, and the date of the last update. Here is an example of metadata:
```
Comment (kwennB) [inline CONTRIBUTING-docs.md:68]: The metadata is essential for successfully building the markdown file. Please include it in the description, as the file will not render locally without it.

Comment (kwennB) [inline CONTRIBUTING-docs.md:68]: There is a whitespace error between the first and second sentences. "_documentation.when_".

Comment (kwennB) [inline CONTRIBUTING-docs.md:68]: It may be best to remove "here is an example of metadata" since you also repeated that phrase in the Example sub-section.

**Review by kwennB — CHANGES_REQUESTED** [review]
Good work, and thank you for your quick response to this issue.

**Comment by kwennB** [comment]
Here is the style guide to also help. https://github.com/json-schema-org/website/blob/main/pages/md-style-guide.md

**Comment by valeriahhdez** [comment]
@techmannih, thanks for your hard work on improving the contribution guide! Your changes have made it clearer and more user-friendly.

I do have a small issue regarding the level of detail in the steps. For straightforward tasks, providing too many steps might come across as condescending. To avoid this, we could simplify the instructions by providing a high-level overview and letting the reader fill in the details.

Let me know if you have any questions.

**Review by valeriahhdez — CHANGES_REQUESTED** [review]
Thank you for your work on this PR!

We don't add detailed step-by-step instructions for straightforward tasks, like adding metadata to a .md file. I suggest adding a general instruction and letting the reader fill in the details themselves.

File: CONTRIBUTING-docs.md [hunk 2]
```diff
@@ -64,6 +64,24 @@ After you have been assigned to an issue, fork the [json-schema-org/website](htt
...
+### Example of Metadata in a New Markdown File:
+Every new `.md` file should start with a YAML front matter block. Here's an example:
+---
+    title: "Your Page Title"
+    section: "docs"  # Can be used to categorize the content
+    date: "YYYY-MM-DD"  # Optional: Date when the page was created or last updated
+    author: "Your Name"  # Optional: Your name if you're the author
+    tags: ["tag1", "tag2"]  # Optional: Tags to categorize the page
+---
+### Steps for Adding Metadata:
```
[the remark this answers is not in the document]
Comment (gregsdennis) [reply CONTRIBUTING-docs.md:80]: I disagree. I think it's nice to know the properties which are used by the rendering engine. To my knowledge they're not standardized, even if there is a common convention.

**Comment by valeriahhdez** [comment]
Looking good @techmannih! 

I left two more comments, but after you addressed them, it will be better to merge

**Comment by valeriahhdez** [comment]
> @valeriahhdez Which comments are still pending? I believe these have already been updated. Could you please revise them again

@techmannih I tagged you in the comments I left a few minutes ago

**Comment by valeriahhdez** [comment]
@techmannih here are the screenshots of the feedback that is pending to be addressed 

Screen Shot 2024-11-29 at 11 28 06
Screen Shot 2024-11-29 at 11 28 34
Screen Shot 2024-11-29 at 11 28 52

**Comment by valeriahhdez** [comment]
LGTM

Good job @techmannih

**Review by valeriahhdez — APPROVED** [review]
LGTM, googd job @techmannih

**Review by kwennB — APPROVED** [review]
LGTM 🚀🚀

**Comment by kwennB** [comment]
@DhairyaMajmudar The docs team approves this PR.

**Review by benjagm — APPROVED** [review]
LGTM

**Pull request merged**